README.md

```diff
@@ -35,686 +43,422 @@ mix.exs
-
-See the [key generation docs](https://hexdocs.pm/jose/key-generation.html) from jose for how to generate your own keys.
-
-To get off the ground quickly, simply replace `<guardian secret key>` in your Guardian config with the output of either:
-
-`$ mix phoenix.gen.secret`
-
-or
-
-`iex(1)> JOSE.JWS.generate_key(%{"alg" => "HS512"}) |> JOSE.JWK.to_map |> elem(1) |> Map.take(["k", "kty"])`
-
-After running `$ mix deps.get` because JOSE is one of Guardian's dependencies.
-
-```elixir
-## Map ##
-
-config :guardian, Guardian,
-  allowed_algos: ["ES512"],
-  secret_key: %{
-    "crv" => "P-521",
-    "d" => "axDuTtGavPjnhlfnYAwkHa4qyfz2fdseppXEzmKpQyY0xd3bGpYLEF4ognDpRJm5IRaM31Id2NfEtDFw4iTbDSE",
-    "kty" => "EC",
-    "x" => "AL0H8OvP5NuboUoj8Pb3zpBcDyEJN907wMxrCy7H2062i3IRPF5NQ546jIJU3uQX5KN2QB_Cq6R_SUqyVZSNpIfC",
-    "y" => "ALdxLuo6oKLoQ-xLSkShv_TA0di97I9V92sg1MKFava5hKGST1EKiVQnZMrN3HO8LtLT78SNTgwJSQHAXIUaA-lV"
-  }
-
-## Tuple ##
-# If, for example, you have your secret key stored externally (in this example, we're using Redix).
-
-# defined elsewhere
-defmodule MySecretKey do
-  def fetch do
-    # Bad practice for example purposes only.
-    # An already established connection should be used and possibly cache the value locally.
-    {:ok, conn} = Redix.start_link
-    rsa_jwk = conn
-      |> Redix.command!(["GET my-rsa-key"])
-      |> JOSE.JWK.from_binary
-    Redix.stop(conn)
-    rsa_jwk
+  def subject_for_token(resource, _claims) do
+    to_string(resource.id)
   end
-end
-
-config :guardian, Guardian,
-  allowed_algos: ["RS512"],
-  secret_key: {MySecretKey, :fetch}
-
-## %JOSE.JWK{} Struct ##
-# Useful if you store your secret key in an encrypted JSON file with the passphrase in an environment variable.
 
-# defined elsewhere
-defmodule MySecretKey do
-  def fetch do
-    System.get_env("SECRET_KEY_PASSPHRASE") |> JOSE.JWK.from_file(System.get_env("SECRET_KEY_FILE"))
+  def resource_from_claims(claims) do
+    find_me_a_resource(claims["sub"])
   end
 end
-
-config :guardian, Guardian,
-  allowed_algos: ["Ed25519"],
-  secret_key: {MySecretKey, :fetch}
 ```
 
-## Serializer
-
-The serializer knows how to encode and decode your resource into and out of the
-token. A simple serializer:
+Add your configuration
 
 ```elixir
-defmodule MyApp.GuardianSerializer do
-  @behaviour Guardian.Serializer
-
-  alias MyApp.Repo
-  alias MyApp.User
-
-  def for_token(user = %User{}), do: { :ok, "User:#{user.id}" }
-  def for_token(_), do: { :error, "Unknown resource type" }
-
-  def from_token("User:" <> id), do: { :ok, Repo.get(User, id) }
-  def from_token(_), do: { :error, "Unknown resource type" }
-end
+config :my_app, MyApp.AuthTokens,
+       issuer: "my_app"
+       secret_key: "Secret key. You can use `mix phx.gen.secret` to get one"
```

Comment:
Let's just copy the secret generator from phoenix and provide `mix guardian.gen.secret`.

https://github.com/phoenixframework/phoenix/blob/master/lib/mix/tasks/phx.gen.secret.ex